IBPSA/Fluid/HeatExchangers/BaseClasses/WetCoilUARated.mo

```diff
@@ -0,0 +1,194 @@
+    "Overall heat transfer coefficient for enthalpy difference";
+  parameter Modelica.SIunits.ThermalConductance UAAir(min=0,start=10,fixed=false)
+    "Air side convective heat transfer coefficient, including fin resistance";
+  parameter Modelica.SIunits.ThermalConductance UAWat(min=0,start=20,fixed=false)
+    "Water side convective heat transfer coefficient";
+  parameter Boolean isFulDry(fixed=false)
+    "Indicator of the fully-dry coil regime";
+  parameter Boolean isFulWet(fixed=false)
+    "Indicator of the fully-wet coil regime";
+  parameter Modelica.SIunits.AbsolutePressure pSatTWatIn=
+    IBPSA.Utilities.Psychrometrics.Functions.saturationPressure(TWatIn)
+    "Saturation pressure of water at the water inlet temperature";
+  parameter Modelica.SIunits.MassFraction X_wSatTWatIn=
+    IBPSA.Utilities.Psychrometrics.Functions.X_pSatpphi(
+      pSat=pSatTWatIn, p=MediumA.p_default, phi=1)
+    "Mass fraction of water in saturated moist air at the water inlet temperature";
+  parameter Modelica.SIunits.SpecificEnthalpy hSatTWatIn=
+    IBPSA.Media.Air.specificEnthalpy_pTX(
+      p=MediumA.p_default, T=TWatIn, X={X_wSatTWatIn,1-X_wSatTWatIn})
+    "Enthalpy of saturated moist air at the water inlet temperature";
+  parameter Modelica.SIunits.AbsolutePressure pSatTWatOut=
+    IBPSA.Utilities.Psychrometrics.Functions.saturationPressure(TWatOut)
+    "Saturation pressure of water at the water oulet temperature";
+  parameter Modelica.SIunits.MassFraction X_wSatTWatOut=
+    IBPSA.Utilities.Psychrometrics.Functions.X_pSatpphi(
+      pSat=pSatTWatOut, p=MediumA.p_default, phi=1)
+    "Mass fraction of water in saturated moist air at the water outlet temperature";
+  parameter Modelica.SIunits.SpecificEnthalpy hSatTWatOut=
+    IBPSA.Media.Air.specificEnthalpy_pTX(
+      p=MediumA.p_default, T=TWatOut, X={X_wSatTWatOut,1-X_wSatTWatOut})
+    "Enthalpy of saturated moist air at the water oulet temperature";
+initial equation
+  isFulDry = if use_Q_flow_nominal then (X_wSatTWatIn >= X_wAirIn) else true;
+  isFulWet = if use_Q_flow_nominal then (X_wSatTWatOut <= X_wAirIn) else true;
+  assert(
+    not use_Q_flow_nominal or
+      hAirOut >= hSatTWatIn and hAirIn >= hSatTWatOut or
+      hAirOut <= hSatTWatIn and hAirIn <= hSatTWatOut,
+    "In " + getInstanceName() +
+    ": The moist air enthalpy at the coil inlet or outlet is unrealistically low. " +
+    "Check the rated conditions.");
+  assert(
+    isFulDry or isFulWet,
+    "In " + getInstanceName() +
+    ": The nominal conditions correspond to a partially-wet coil regime. " +
+    "The modeling uncertainty under such conditions has not been assessed. " +
+    "Rather specify nominal conditions in fully-dry or fully-wet regime.",
+    level=AssertionLevel.warning);
+  if use_Q_flow_nominal then
+    if isFulDry then
+      TAirOut = TAirIn + QTot_flow / mAir_flow / cpAir;
+      LMED=IBPSA.Fluid.HeatExchangers.BaseClasses.lmtd(
```

Comment:
The function `IBPSA.Fluid.HeatExchangers.BaseClasses.lmtd` does not support ∆T1=∆T2 and the implementation could guard for this specific case by adding an if-else.

Reply:
Addressed considering `lmtd(dT1, dT2) ~ dT1` when `dT1 ~ dT2`.

Reply:
Where is that in the code? I don't see it.

Reply:
Hmm... This is here: https://github.com/ibpsa/modelica-ibpsa/pull/1549/commits/38f7a3452d8c8bc5d4c6e5dccbe96c6b75e150ab#diff-21446cd2c6a597c96aeeeea85dde48c6ba41c19736a873dbf07b850445325a6fR129